src/collection.ts

```diff
@@ -975,12 +998,111 @@ export class Collection<TSchema extends Document = Document> {
+  /**
+   * @internal
+   *
+   * Returns all search indexes for the current collection.
+   *
+   * @param name - The name of the index to search for.  Only indexes with matching index names will be returned.
+   * @param options - The options for the list indexes operation.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   */
+  listSearchIndexes(name: string, options?: ListSearchIndexesOptions): ListSearchIndexesCursor;
+  listSearchIndexes(
+    indexNameOrOptions?: string | ListSearchIndexesOptions,
+    options?: ListSearchIndexesOptions
+  ): ListSearchIndexesCursor {
+    options =
+      typeof indexNameOrOptions === 'object' ? indexNameOrOptions : options == null ? {} : options;
+    const indexName =
+      indexNameOrOptions == null
+        ? null
+        : typeof indexNameOrOptions === 'object'
+        ? null
+        : indexNameOrOptions;
+
+    return new ListSearchIndexesCursor(this as TODO_NODE_3286, indexName, options);
+  }
+
+  /**
+   * @internal
+   *
+   * Creates a single search index for the collection.
+   *
+   * @param description - The index description for the new search index.
+   * @returns A promise that resolves to the name of the new search index.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   */
+  async createSearchIndex(description: SearchIndexDescription): Promise<string> {
+    const [index] = await this.createSearchIndexes([description]);
+    return index;
+  }
+
+  /**
+   * @internal
+   *
+   * Creates multiple search indexes for the current collection.
+   *
+   * @param descriptions - An array of `SearchIndexDescription`s for the new search indexes.
+   * @returns A promise that resolves to an array of the newly created search index names.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   * @returns
+   */
+  async createSearchIndexes(descriptions: SearchIndexDescription[]): Promise<string[]> {
+    return executeOperation(
+      this.client,
+      new CreateSearchIndexesOperation(this as TODO_NODE_3286, descriptions)
+    );
+  }
+
+  /**
+   * @internal
+   *
+   * Deletes a search index by index name.
+   *
+   * @param name - The name of the search index to be deleted.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   */
+  async dropSearchIndex(name: string): Promise<void> {
+    return executeOperation(
+      this.client,
+      new DropSearchIndexOperation(this as TODO_NODE_3286, name)
+    );
+  }
+
+  /**
+   * @internal
+   *
+   * Updates a search index by replacing the existing index definition with the provided definition.
+   *
+   * @param name - The name of the search index to update.
+   * @param definition - The new search index definition.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   */
+  async updateSearchIndex(name: string, definition: Document): Promise<void> {
```

Comment:
Have we enumerated everything that we'll need to remove `@internal` tags from when we are ready to release the feature? And if so, which ticket documents it?

Reply:
[NODE-5319](https://jira.mongodb.org/browse/NODE-5319) has been filed.